dev/PagerControl/PagerControl.cpp

```diff
@@ -0,0 +1,776 @@
+    SetValue(s_TemplateSettingsProperty, templateSettings);
+
+    SetDefaultStyleKey(this);
+}
+
+PagerControl::~PagerControl()
+{
+    m_rootGridKeyDownRevoker.revoke();
+    m_comboBoxSelectionChangedRevoker.revoke();
+    m_firstPageButtonClickRevoker.revoke();
+    m_previousPageButtonClickRevoker.revoke();
+    m_nextPageButtonClickRevoker.revoke();
+    m_lastPageButtonClickRevoker.revoke();
+}
+
+void PagerControl::OnApplyTemplate()
+{
+    if (PrefixText().empty())
+    {
+        PrefixText(ResourceAccessor::GetLocalizedStringResource(SR_PagerControlPrefixTextName));
+    }
+    if (SuffixText().empty())
+    {
+        SuffixText(ResourceAccessor::GetLocalizedStringResource(SR_PagerControlSuffixTextName));
+    }
+    if (const auto rootGrid = GetTemplateChildT<winrt::FrameworkElement>(c_rootGridName, *this))
+    {
+        m_rootGridKeyDownRevoker = rootGrid.KeyDown(winrt::auto_revoke, { this, &PagerControl::OnRootGridKeyDown });
+    }
+
+    if (const auto firstPageButton = GetTemplateChildT<winrt::Button>(c_firstPageButtonName, *this))
+    {
+        m_firstPageButtonClickRevoker = firstPageButton.Click(winrt::auto_revoke, { this, &PagerControl::FirstButtonClicked });
+    }
+    if (const auto previousPageButton = GetTemplateChildT<winrt::Button>(c_previousPageButtonName, *this))
+    {
+        m_previousPageButtonClickRevoker = previousPageButton.Click(winrt::auto_revoke, { this, &PagerControl::PreviousButtonClicked });
+    }
+    if (const auto nextPageButton = GetTemplateChildT<winrt::Button>(c_nextPageButtonName, *this))
+    {
+        m_nextPageButtonClickRevoker = nextPageButton.Click(winrt::auto_revoke, { this, &PagerControl::NextButtonClicked });
+    }
+    if (const auto lastPageButton = GetTemplateChildT<winrt::Button>(c_lastPageButtonName, *this))
+    {
+        m_lastPageButtonClickRevoker = lastPageButton.Click(winrt::auto_revoke, { this, &PagerControl::LastButtonClicked });
+    }
+
+    if (const auto comboBox = GetTemplateChildT<winrt::ComboBox>(c_comboBoxName, *this))
+    {
+        m_comboBox.set(comboBox);
+        comboBox.SelectedIndex(SelectedPageIndex() - 1);
+        winrt::AutomationProperties::SetName(comboBox, ResourceAccessor::GetLocalizedStringResource(SR_PagerControlPageTextName));
+        m_comboBoxSelectionChangedRevoker = comboBox.SelectionChanged(winrt::auto_revoke, { this, &PagerControl::ComboBoxSelectionChanged });
+    }
+    if (const auto numberBox = GetTemplateChildT<winrt::NumberBox>(c_numberBoxName, *this))
+    {
+        m_numberBox.set(numberBox);
+        numberBox.Value(SelectedPageIndex() + 1);
+        winrt::AutomationProperties::SetName(numberBox, ResourceAccessor::GetLocalizedStringResource(SR_PagerControlPageTextName));
+        m_numberBoxValueChangedRevoker = numberBox.ValueChanged(winrt::auto_revoke, { this,&PagerControl::NumberBoxValueChanged });
+    }
+    if (const auto repeater = GetTemplateChildT<winrt::ItemsRepeater>(c_numberPanelRepeaterName, *this))
```

Comment:
lambda isn't needed here since we don't use the local more than once, just move this call into the member variable setter.